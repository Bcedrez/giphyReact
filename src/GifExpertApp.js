import React, { useState } from "react";
import { AddCatergory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

// const categorias = ['One Punch', 'Samurai X', 'Dragon Ball'];



const GifExpertApp = ()=>{

    const [categorias, setCategorias] = useState(['Samurai X'])

    return(
        <>
            <h2>GifExpertApp</h2>
            <hr/>

            <AddCatergory setCategorias={setCategorias}/>

            <ol>
                {
                    categorias.map((categoria,i) =>{
                        return <GifGrid
                                        key={categoria}                                      
                                        categoria={categoria} />
                    })
                }
            </ol>



        </>
    )

}

export default GifExpertApp