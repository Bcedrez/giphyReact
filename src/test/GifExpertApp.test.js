import React from 'react';
import GifExpertApp from '../GifExpertApp';
const { shallow } = require("enzyme");

describe('Pruebas a componente <GifExpertApp />',()=>{


    test('Renderizado correcto',()=>{

        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar lista de categorias',()=>{

        const categories = ['Prueba','test'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })

})


