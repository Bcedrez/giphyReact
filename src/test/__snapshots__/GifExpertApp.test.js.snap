// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a componente <GifExpertApp /> Renderizado correcto 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCatergory
    setCategorias={[Function]}
  />
  <ol />
</Fragment>
`;
